lib/std/mem.zig

```diff
@@ -105,6 +105,23 @@ pub const Allocator = struct {
         return self.alignedAlloc(T, null, n);
     }
 
+    /// Takes a sentinel slice type `T` and allocates an array of `n + 1` items of
+    /// the child type of `T`.  All items are `undefined` except the last which
+    /// is set to the sentinel value of `T`. Depending on the Allocator implementation,
+    /// it may be required to call `free` once the memory is no longer needed, to
+    /// avoid a resource leak. If the `Allocator` implementation is unknown,
+    /// then correct code will call `free` when done.
+    ///
+    /// For allocating a single item, see `create`.
+    pub fn allocSentinel(self: *Allocator, comptime T: type, n: usize) Error!T {
+        const info = @typeInfo(T).Pointer;
+        const sentinel = info.sentinel orelse
```

Comment:
Would be nice to use the `Sentinel` function from https://github.com/ziglang/zig/pull/3972/files#diff-68e1075f555c39ac8519c599ce5c73e6R120 here.

Reply:
If the slice type does not have a sentinel value, here's the error I get with the current version:
```bash
./lib/zig/std/mem.zig:119:13: error: allocSentinel requires a slice type with a sentinel value
            @compileError("allocSentinel requires a slice type with a sentinel value");
            ^
/home/marler8997/git/zig/test.zig:4:58: note: called from here
    const mem = try std.heap.page_allocator.allocSentinel([]u8, 10);
                                                         ^
/home/marler8997/git/zig/test.zig:3:29: note: called from here
pub fn main() anyerror!void {
                            ^

```

and here's the error code I get if I use `std.meta.Sentinel`:
```bash
./lib/zig/std/meta.zig:128:53: error: unable to unwrap null
                    return comptime ptrInfo.sentinel.?;
                                                    ^
./lib/zig/std/mem.zig:120:48: note: called from here
        comptime const sentinel = meta.Sentinel(T);
                                               ^
/home/marler8997/git/zig/test.zig:4:58: note: called from here
    const mem = try std.heap.page_allocator.allocSentinel([]u8, 10);
                                                         ^
/home/marler8997/git/zig/test.zig:3:29: note: called from here
pub fn main() anyerror!void {
                            ^
./lib/zig/std/mem.zig:120:48: note: referenced here
        comptime const sentinel = meta.Sentinel(T);
                                               ^
```

The error message seems harder to decipher.  What's the benefit we get using `meta.Sentinel` at the cost of having a worse compile error message when we don't have a sentinel value?

Reply:
resolving since this no longer applies because the sentinel value is now passed in as an argument